components/TodoList.vue

```diff
@@ -0,0 +1,69 @@
+<template>
+  <div>
+    <label for="filter">Фильтр</label>
+      <input placeholder="filter..." v-model="filter" id="filter"/>
+      <TransactionPaginated
+        :total="total"
+        :currentPage="currentPage"
+        @changePage="changePage"
+        @moveToNext="changePage"
+        @moveToBack="changePage"
+      />
+    <h3>Todo Name</h3>
+    <table>
+        <tr v-for="todo in filteredTodos" :key="todo.id">
+          <td>{{ todo.title }}</td>
+        </tr>
+    </table>
+  </div>
+</template>
+
+<script>
+import TransactionPaginated from "./TransactionPaginated.vue";
+export default {
+  props: {
+    todos: { type: Array, required: true },
+  },
+  data() {
+    return {
+      filter: "",
+      currentPage: 1,
+      paginatedTodo: null,
+      hasNext: true,
+    };
+  },
+  components: {
+    TransactionPaginated,
+  },
+  computed: {
+    total() {
+      return this.filter ? this.paginatedTodo.length : this.todos.length;
+    },
+    filteredTodos() {
+      const start = (this.currentPage - 1) * 25;
+      const end = this.currentPage * 25;
+      this.paginatedTodo = this.filter
+        ? this.todos
+            .filter((todo) =>
+              [todo.title].some((text) =>
```

Comment:
simplify this 
![image](https://user-images.githubusercontent.com/72648845/186086137-d8bdbd89-9046-48d4-b3f5-ca4fb1207163.png)